zerver/tests/test_project.py

```diff
@@ -0,0 +1,30 @@
+import importlib
+from unittest import mock
+
+from zerver.lib.test_classes import ZulipTestCase
+from zproject import config
+
+
+class ConfigTest(ZulipTestCase):
+    def test_get_mandatory_secret_succeed(self) -> None:
+        secret = config.get_mandatory_secret("shared_secret")
+        self.assertGreater(len(secret), 0)
+
+    def test_get_mandatory_secret_failed(self) -> None:
+        with self.assertRaises(config.ZulipSettingsError) as err:
+            config.get_mandatory_secret("nonexistent")
+        self.assertIn("nonexistent", str(err.exception))
```

Comment:
`assertRaisesRegex` is probably a clearer alternative to a separate `assertIn` check`.